packages/playground/src/components/loaders/skeleton-list.tsx

```diff
@@ -1,17 +1,32 @@
-import { Skeleton, StackedList } from '@harnessio/canary'
-import React from 'react'
+import { cn, Skeleton, StackedList } from '@harnessio/canary'
+import React, { useEffect, useState } from 'react'
 
 // Helper function to generate random percentage width within a range
 const getRandomPercentageWidth = (min: number, max: number) => `${Math.floor(Math.random() * (max - min + 1)) + min}%`
 
 // Helper function to generate random pixel width within a range
 const getRandomPixelWidth = (min: number, max: number) => `${Math.floor(Math.random() * (max - min + 1)) + min}px`
 
-export const SkeletonList = () => {
+interface SkeletonListProps {
+  delay?: number
+}
+
+export const SkeletonList = ({ delay = 300 }: SkeletonListProps) => {
   const listItems = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 
+  let startDelayTimer: number
+  const [visible, setVisible] = useState(false)
+  useEffect(() => {
+    startDelayTimer = setTimeout(() => {
+      setVisible(true)
+    }, delay)
+    return () => {
+      clearTimeout(startDelayTimer)
+    }
+  }, [])
+
   return (
-    <div className="relative w-full h-full">
+    <div className={cn('relative w-full h-full transition-opacity', { ['opacity-0']: !visible })}>
```

Comment:
Also perhaps needs an easing function and duration for the transition, e.g. 'duration-100 ease-in-out'. Actually perhaps you don't need the timeout stuff at all since you can set the transition-delay in tailwind.

Reply:
eg. 

`<div
      className={cn(
        'relative w-full h-full opacity-0 transition-opacity duration-150',
        `delay-${delay}`,
        'opacity-100'
      )}
    >`
    
 Only thing to check via typing is that delay value is an preset tailwind delay-X value

Reply:
Actually no, you will need a state approach otherwise the opacity will always be 100% using pure tailwind. Unless you use an animation with start opacity-0 and end opacity-100 and use forwards.

Reply:
yes, still need state, but removed the timer